crates/store/re_server/src/store/dataset.rs

```diff
@@ -86,47 +98,49 @@ impl Dataset {
         partition_ids: &'a [PartitionId],
     ) -> Result<impl Iterator<Item = (&'a PartitionId, &'a Partition)>, Error> {
         if partition_ids.is_empty() {
-            Ok(Either::Left(self.partitions.iter()))
+            Ok(Either::Left(self.inner.partitions.iter()))
         } else {
             // Validate that all partition IDs exist
             for id in partition_ids {
-                if !self.partitions.contains_key(id) {
+                if !self.inner.partitions.contains_key(id) {
                     return Err(Error::PartitionIdNotFound(id.clone(), self.id));
                 }
             }
 
             Ok(Either::Right(partition_ids.iter().filter_map(|id| {
-                self.partitions.get(id).map(|partition| (id, partition))
+                self.inner
+                    .partitions
+                    .get(id)
+                    .map(|partition| (id, partition))
             })))
         }
     }
 
     pub fn set_dataset_details(&mut self, details: DatasetDetails) {
-        self.details = details;
-        self.updated_at = jiff::Timestamp::now();
+        self.inner.modify().details = details;
```

Comment:
Similarly to the `set_name` shouldn't it be trivial to check whether the new details are actually _new_? In the case where they are, this will update the `last_updated_timestamp` even though nothing changed.

`DatasetDetails` can implement `PartialEq` and `Eq`, all its components already do.

Reply:
good call